addons/l10n_ec/migrations/3.4/post-migration.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
```

Comment:
@jco-odoo to be checked
cc' @pepetreshere

Reply:
@jco-odoo @odoostan  this is still to be reviewed, as probably this file should be removed, but not sure if you can check with the migration team if this could help
cc' @ren-odoo @fvz-odoo

Reply:
I can review the script but I don't know whether the file should be removed or not.
I can ask anything to the migration team, I just don't know what to ask.
Why should the file be removed ?

Reply:
@odoostan if you can ask the migration team what they think or someone from the team it will be helpful.
We can probably keep with this, it's just that we usually don't do this with other partners when they develop something (they don't do the migration script), but it doesn't mean that we need to remove it.

Thanks!
cc' @ren-odoo @fvz-odoo @pepetreshere

Reply:
We can do something like this, because it can help some users (although I don't think there will be many), but we should keep it to the minimum necessary.

Reply:
@jco-odoo I've reviewed how to keep it to the minimum necessary:
- [x] update_withhold_type(env), should be kept, otherwise withhold taxes won't be selectable in withhold form
- [x] update_type_tax_use(env), should be kept, otherwise withhold taxes won't be selectable in withhold form
- [x] unlink_old_withhold_group(env), should be kept, although the old record is marked as noupdate=True so it will remain there and might cause a small confusion to users when manually adding new taxes and the field l10n_ec_type won't have an selectable value that makes sense.
- [ ] update_vat_withhold_base_percent(env), optional, the withhold wizard method _onchange_base() doesn't care about the base percent configured in the withhold tax at all (it ignores the tax base, and uses the base comming from the invoice)
- [ ] recompute_invoice_names(env), not really sure, the current implementation of _check_unique_vendor_number() requires the  "name" to be properly concatenated with the document code to work, so if we change the code (as we did) then we need to recompute the name, so the method would be necessary. I would also remark that the code need to be changed because previously several different document types had the same code!. (Please check method _check_unique_vendor_number(), it has a comment that says "# by validating name we validate l10n_latam_document_type_id").
- [ ] change the withhold tax amount from negative to positive? this is not implemented as it is not necesary and because we agreed to let it negative to limit the disruption to custom modules built on top of the old version. I would prefer to have them as positive as the other changes are already important. I just bring this here so you have the whole context on the discussed migration scripts.

Reply:
@jco-odoo most of them was already deleted in older commits. Now I have deleted the third one regarding the unlink of the old withhold groups.

Reply:
Changing the factor_percent to 100% in vat withhold is not strictly needed, the compute_all is not used to compute the base amount, and the concept doesn't apply to withholds. The base amount is either summed up from the invoices lines on the _suggest_withhold_lines() or set in the onchange method for manual lines and even so the user might need to change the base amount.

@jco-odoo this was discussed and agreed not to change. But if you ask me, I would like to change it for a different reason: aesthetics. If we go for aesthetics I would also change the tax percentage in all withholds from negative to positive (negative was an old trick to do it all together with the purchase invoice)

I would need your confirmation on your final decision (you used a "would" so it sounded like a thought to me)

Reply:
@pepetreshere I checked in the compute_all method and the factor_percent is used, at least the one on the tax part and that is the one we are changing.  And, as far as I know we use the compute_all method now (we might even still change) to calculate the withholding amounts.  

To have it as a negative tax still makes sense to me, because you are still deducing the paid amount.  (it depends of course on the perspective)

Reply:
@jco-odoo yes, you are right, in the older implementation (before using the suggested compute_all method it was working). I've added back the migration script for this part, if you agree let's close the comment on the migration script.

Reply:
@pepetreshere should change it to SQL

Reply:
@jco-odoo it is done in [this commit](https://github.com/odoo/odoo/pull/92472/commits/94811b2dc2e2319ea19f6d7553b0a35ff65c8c1e), I think we could finally close the comment